News does not work

I'm not sure how to properly configure an installed snap to enable network access.

I'd like this most excellent snap to support the News feature in SPD.  I wind up reading the app news from my Android version of the game.

Thanks!

This could be an SSL/cert issue, as I am seeing this:

```
shattered-pixel-dungeon
DRM_IOCTL_ASAHI_GET_PARAMS failed: Bad address
glx: failed to create dri3 screen
failed to load driver: asahi
DRM_IOCTL_ASAHI_GET_PARAMS failed: Bad address
glx: failed to create dri3 screen
failed to load driver: asahi
[Controllers] added manager for application, 1 managers active
[GAME] javax.net.ssl.SSLException: (internal_error) Unexpected error: java.security.InvalidAlgorithmParameterException: the trustAnchors parameter must be non-empty
        at java.base/sun.security.ssl.Alert.createSSLException(Alert.java:132)
        at java.base/sun.security.ssl.TransportContext.fatal(TransportContext.java:383)
        at java.base/sun.security.ssl.TransportContext.fatal(TransportContext.java:326)
        at java.base/sun.security.ssl.TransportContext.fatal(TransportContext.java:321)
        at java.base/sun.security.ssl.SSLSocketImpl.handleException(SSLSocketImpl.java:1708)
        at java.base/sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:470)
        at java.base/sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:426)
        at java.base/sun.net.www.protocol.https.HttpsClient.afterConnect(HttpsClient.java:586)
        at java.base/sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.connect(AbstractDelegateHttpsURLConnection.java:187)
        at java.base/sun.net.www.protocol.https.HttpsURLConnectionImpl.connect(HttpsURLConnectionImpl.java:141)
        at com.badlogic.gdx.net.NetJavaImpl$2.run(NetJavaImpl.java:223)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:572)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
        at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: java.lang.RuntimeException: Unexpected error: java.security.InvalidAlgorithmParameterException: the trustAnchors parameter must be non-empty
        at java.base/sun.security.validator.PKIXValidator.<init>(PKIXValidator.java:97)
        at java.base/sun.security.validator.Validator.getInstance(Validator.java:173)
        at java.base/sun.security.ssl.X509TrustManagerImpl.getValidator(X509TrustManagerImpl.java:308)
        at java.base/sun.security.ssl.X509TrustManagerImpl.checkTrustedInit(X509TrustManagerImpl.java:183)
        at java.base/sun.security.ssl.X509TrustManagerImpl.checkTrusted(X509TrustManagerImpl.java:198)
        at java.base/sun.security.ssl.X509TrustManagerImpl.checkServerTrusted(X509TrustManagerImpl.java:132)
        at java.base/sun.security.ssl.CertificateMessage$T13CertificateConsumer.checkServerCerts(CertificateMessage.java:1271)
        at java.base/sun.security.ssl.CertificateMessage$T13CertificateConsumer.onConsumeCertificate(CertificateMessage.java:1164)
        at java.base/sun.security.ssl.CertificateMessage$T13CertificateConsumer.consume(CertificateMessage.java:1107)
        at java.base/sun.security.ssl.SSLHandshake.consume(SSLHandshake.java:393)
        at java.base/sun.security.ssl.HandshakeContext.dispatch(HandshakeContext.java:477)
        at java.base/sun.security.ssl.HandshakeContext.dispatch(HandshakeContext.java:448)
        at java.base/sun.security.ssl.TransportContext.dispatch(TransportContext.java:206)
        at java.base/sun.security.ssl.SSLTransport.decode(SSLTransport.java:172)
        at java.base/sun.security.ssl.SSLSocketImpl.decode(SSLSocketImpl.java:1506)
        at java.base/sun.security.ssl.SSLSocketImpl.readHandshakeRecord(SSLSocketImpl.java:1421)
        at java.base/sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:455)
        ... 10 more
Caused by: java.security.InvalidAlgorithmParameterException: the trustAnchors parameter must be non-empty
        at java.base/java.security.cert.PKIXParameters.setTrustAnchors(PKIXParameters.java:200)
        at java.base/java.security.cert.PKIXParameters.<init>(PKIXParameters.java:120)
        at java.base/java.security.cert.PKIXBuilderParameters.<init>(PKIXBuilderParameters.java:104)
        at java.base/sun.security.validator.PKIXValidator.<init>(PKIXValidator.java:94)
        ... 26 more

[GAME] @@ You descend to floor 1 of the dungeon.
```

--- a/snap/snapcraft.yaml
+++ b/snap/snapcraft.yaml
@@ -64,10 +64,10 @@ parts:
     after: [alsa-pulseaudio,launcher]
     build-packages:
       - wget
-    stage-packages:
-      - openjdk-21-jre-headless
       - ca-certificates
       - ca-certificates-java
+    stage-packages:
+      - openjdk-21-jre-headless
     plugin: nil
     override-build: |
       mkdir -p "$SNAPCRAFT_PART_INSTALL/bin"
@@ -75,17 +75,16 @@ parts:
     override-prime: |
       craftctl default
       # openjdk ships cacerts and blacklisted.certs as symlinks pointing into
-      # /etc outside the snap, which the store linter rejects. Replace each
-      # symlink with a real copy of the file from within the prime tree.
+      # /etc outside the snap, which the store linter rejects. ca-certificates-java
+      # is a post-install hook package so its files live on the host, not in the
+      # prime tree. Find all such external symlinks and replace them with real
+      # copies sourced directly from the host filesystem.
       find usr/lib/jvm -name "cacerts" -o -name "blacklisted.certs" | while read link; do
         if [ -L "$link" ]; then
-          # Resolve relative to the prime directory, not the host filesystem
-          target=$(readlink "$link")
-          # Strip leading / to make it relative to prime tree
-          target_rel="${target#/}"
-          if [ -f "$target_rel" ]; then
+          target=$(readlink -f "$link")
+          if [ -f "$target" ]; then
             rm "$link"
-            cp "$target_rel" "$link"
+            cp "$target" "$link"
           fi
         fi
       done

--- a/snap/snapcraft.yaml
+++ b/snap/snapcraft.yaml
@@ -72,3 +72,17 @@ parts:
     override-build: |
       mkdir -p "$SNAPCRAFT_PART_INSTALL/bin"
       wget -O "$SNAPCRAFT_PART_INSTALL/bin/Shattered_Pixel_Dungeon.jar" "https://github.com/00-Evan/shattered-pixel-dungeon/releases/download/$SNAPCRAFT_PROJECT_VERSION/ShatteredPD-$SNAPCRAFT_PROJECT_VERSION-Java.jar"
+    override-prime: |
+      craftctl default
+      # Replace external symlinks with actual file copies so the snap passes store review.
+      # openjdk on core22 ships cacerts and blacklisted.certs as symlinks into /etc,
+      # which the store linter rejects as external symlinks.
+      for link in usr/lib/jvm/java-*/lib/security/cacerts usr/lib/jvm/java-*/lib/security/blacklisted.certs; do
+        if [ -L "$link" ]; then
+          target=$(readlink -f "$link" 2>/dev/null || true)
+          if [ -f "$target" ]; then
+            rm "$link"
+            cp "$target" "$link"
+          fi
+        fi
+      done

--- a/snap/snapcraft.yaml
+++ b/snap/snapcraft.yaml
@@ -74,15 +74,18 @@ parts:
       wget -O "$SNAPCRAFT_PART_INSTALL/bin/Shattered_Pixel_Dungeon.jar" "https://github.com/00-Evan/shattered-pixel-dungeon/releases/download/$SNAPCRAFT_PROJECT_VERSION/ShatteredPD-$SNAPCRAFT_PROJECT_VERSION-Java.jar"
     override-prime: |
       craftctl default
-      # Replace external symlinks with actual file copies so the snap passes store review.
-      # openjdk on core22 ships cacerts and blacklisted.certs as symlinks into /etc,
-      # which the store linter rejects as external symlinks.
-      for link in usr/lib/jvm/java-*/lib/security/cacerts usr/lib/jvm/java-*/lib/security/blacklisted.certs; do
+      # openjdk ships cacerts and blacklisted.certs as symlinks pointing into
+      # /etc outside the snap, which the store linter rejects. Replace each
+      # symlink with a real copy of the file from within the prime tree.
+      find usr/lib/jvm -name "cacerts" -o -name "blacklisted.certs" | while read link; do
         if [ -L "$link" ]; then
-          target=$(readlink -f "$link" 2>/dev/null || true)
-          if [ -f "$target" ]; then
+          # Resolve relative to the prime directory, not the host filesystem
+          target=$(readlink "$link")
+          # Strip leading / to make it relative to prime tree
+          target_rel="${target#/}"
+          if [ -f "$target_rel" ]; then
             rm "$link"
-            cp "$target" "$link"
+            cp "$target_rel" "$link"
           fi
         fi
       done

--- a/snap/snapcraft.yaml
+++ b/snap/snapcraft.yaml
@@ -66,9 +66,9 @@ parts:
       - wget
     stage-packages:
       - openjdk-21-jre-headless
+      - ca-certificates
+      - ca-certificates-java
     plugin: nil
     override-build: |
       mkdir -p "$SNAPCRAFT_PART_INSTALL/bin"
       wget -O "$SNAPCRAFT_PART_INSTALL/bin/Shattered_Pixel_Dungeon.jar" "https://github.com/00-Evan/shattered-pixel-dungeon/releases/download/$SNAPCRAFT_PROJECT_VERSION/ShatteredPD-$SNAPCRAFT_PROJECT_VERSION-Java.jar"
-    prime:
-      - -usr/lib/jvm/java-*/lib/security/cacerts

